common/components/org.wso2.carbon.analytics.util/src/main/java/org/wso2/carbon/analytics/common/spark/udf/DateTimeUDF.java

```diff
@@ -44,6 +44,13 @@ public Integer getMonth(Long timeStamp) throws ParseException {
         }
     }
 
+    public Integer getWeek(Long timeStamp) throws ParseException{
+        synchronized (cal){
```

Comment:
Hi Chandana,

These code fixed are done in commits https://github.com/wso2/shared-analytics/pull/4/files#diff-1537c24d7a186cdcb0f302c00473f5c5R48

Thanks,
SajithD